### PR TITLE
Add `whitebox.wd` + package options update

### PR DESCRIPTION
**Pull request by brownag**
A continuation of "package options" theme. 

I have bundled bug fixes with new features, doc improvements, and more testing as I have tried more edge cases

 - general improvements in consistent option getting and setting
 - new tests of `wbt_init()` without a valid `exe_path` set (can run on CRAN); and better messages
 - added `"all"` option to possibilities for `whitebox.verbose` so you can globally trigger `verbose_mode=TRUE` for multiline output from every tool. If "all" `wbt_verbose()` will return TRUE (still always logical).
   - Use `wbt_verbose("all")` or `wbt_init(..., verbose = "all")` 
   - Most lines of output in these cases are percent progress, but some tools do have interesting messages produced and it might (?) be relevant to debugging
 - new tests of working directories (setting/changing them, and "unsetting" them)
    * WhiteboxTools working directory controlled with (`R_WHITEBOX_WD`) or package option (`whitebox.wd`) 
        * Default does nothing different; current `wd` arguments passed directly to functions are respected/prioritized
        * Get / set a session working directory with `wbt_wd()`/`wbt_init(wd=...)`
        * "Unset" with `wbt_wd("")` or `wbt_wd(NA)`. This will set the package option temporarily to your current R working directory, _then_ after next successful tool run the WhiteboxTools directory updates and the R package unsets the option to `""` (so that the `--wd` flag is dropped). For other directories set the path with `wd` argument.
        * This working directory routine could probably somehow make use of the settings.json that WhiteboxTools produces, but at this point it is not too sophisticated and uses only user-set paths and `getwd()` (which is system() default)
 - fixing some of the logic of system environment variable overrides not triggering correctly

I was able to successfully build hydroweight README w/ https://github.com/giswqs/whiteboxR/commit/c695a8a57f2e16eab80f2da72b98be5090a76b10

### CODE REVIEW

File: R/wbt.R [hunk 1]
```diff
@@ -77,24 +173,34 @@ wbt_options <- function(exe_path = NULL, verbose = NULL) {
...
+  # NA is treated NULL (no effect)
+  if (length(verbose) != 1 || is.na(verbose)) {
+    verbose <- NULL
+  }
   
-  # system environment var takes precedence, but defaults FALSE
-  sysverbose <- Sys.getenv("R_WHITEBOX_VERBOSE", unset = FALSE)
-  if (sysverbose) {
+  # system environment var takes precedence
+  sysverbose <- Sys.getenv("R_WHITEBOX_VERBOSE")
```
Comment (brownag) [inline R/wbt.R:182]: 1. lacks support for `"all"` via system option
2. revisit logic around NA/logical conversion of character `sysverbose`

Comment (brownag) [reply R/wbt.R]: 9feb179 gets the behavior I wanted for this option

``` r
library(whitebox)

Sys.setenv("R_WHITEBOX_VERBOSE" = TRUE)
print(wbt_verbose(FALSE))
#> [1] TRUE

Sys.setenv("R_WHITEBOX_VERBOSE" = "all")
print(wbt_verbose(FALSE))
#> [1] TRUE

Sys.setenv("R_WHITEBOX_VERBOSE" = FALSE)
print(wbt_verbose(TRUE))
#> [1] FALSE

Sys.setenv("R_WHITEBOX_VERBOSE" = 0)
print(wbt_verbose(TRUE))
#> [1] FALSE

Sys.setenv("R_WHITEBOX_VERBOSE" = 1)
print(wbt_verbose(FALSE))
#> [1] TRUE
```

**Review by brownag — COMMENTED** [review]
The updates to package options in general here need to be merged, and the working directory functionality has some decent testing behind it so it shouldn't hurt anything to add it. 

There are probably some edge cases that can be confusing if doing a lot of changing working directories but I find it to be stable and useful when I set a single `wd` and use it for a R session.
